proxy/src/inbound.rs

```diff
@@ -95,14 +83,15 @@ where
     ///
     /// Buffering is currently unbounded and does not apply timeouts. This must be
     /// changed.
-    fn bind_service(&mut self, addr: &SocketAddr) -> Result<Self::Service, Self::RouteError> {
-        debug!("building inbound client to {}", addr);
+    fn bind_service(&mut self, key: &Self::Key) -> Result<Self::Service, Self::RouteError> {
+        let &(ref addr, proto) = key;
```

Comment:
TIOLI, but why not:
```rust
fn bind_service(&mut self, &(ref addr, proto): &Self::Key) -> Result<Self::Service, Self::RouteError> {
    ...
```

Reply:
I personally find destructuring in method signatures to be distracting in most cases.

Reply:
That's fair; I find not introducing intermediate identifiers slightly clearer, but it's a matter of taste!